core/ArchiveProcessor/Loader.php

```diff
@@ -63,20 +64,30 @@ protected function mustProcessVisitCount($visits)
 
     public function prepareArchive($pluginName)
     {
-        $this->params->setRequestedPlugin($pluginName);
+        $originalIdSite = Common::getRequestVar('idSite', false);
```

Comment:
BTW we may need to be careful here since this will read from `$_GET+$_POST`. I'm thinking of possible edge cases where an attacker might send a different value for `$_GET` and `$_POST` and I wonder if this could result in random outcomes or accessing data later after the restoring of `originalIdSite` for example. I haven't thought it through but there could be certainly cases I suppose depending on the application code that this might be a problem. Also the reset would maybe not work or not be applied since it wouldn't overwrite a posted idsite.